tests/tests_api/test_group.py

```diff
@@ -477,3 +477,75 @@ def test_dashboard_group_confirm_ready_to_start_general_check(
             log_event: TeacherLogEvent = TeacherLogEvent.objects.get(teacher_id=teacher.pk)
             assert log_event.type == TeacherLogEventType.GROUP_CONFIRMED
             assert_date_time_with_timestamp(log_event.date_time, timestamp)
+
+
+class TestDashboardGroupDiscard:
+    @pytest.mark.parametrize(
+        "base_path",
+        [
+            "/api/dashboard/groups",
+            "/api/groups",
+        ],
+    )
+    def test_dashboard_group_discard_general_check(
+        self, api_client, pending_group, timestamp, base_path
+    ):
+        group_students, group_teachers, group_coordinators = (
+            list(pending_group.students.iterator()),
+            list(pending_group.teachers.iterator()),
+            list(pending_group.coordinators.iterator()),
+        )
+        students_group_count = [student.groups.count() - 1 for student in group_students]
+        teachers_group_count = [teacher.groups.count() - 1 for teacher in group_teachers]
+        coordinators_group_count = [
+            coordinator.groups.count() - 1 for coordinator in group_coordinators
+        ]
+        discard_reason = GroupDiscardReason.NOT_ENOUGH_STUDENTS
+        response = api_client.delete(
+            f"{base_path}/{pending_group.id}/discard/?discard_reason={discard_reason}"
```

Comment:
It's better to pass params with special argument, not as a string, kind of

`api_client.delete(path, {"discard_reason": discard_reason"})`

Reply:
Seems like it does not work for `delete`, I tried data and params as well. Maybe it is a reason to change method, not sure

Reply:
Strange. But okay.
Also, I agree with DELETE method if it really deleting instance. If we will decide not to delete it completely - let's change it to POST

Reply:
I think that tentative groups can be deleted... Maybe that requires further thinking. Or maybe we should just vote :)